File: lib/services/publicaciones_service.dart
import 'package:dio/dio.dart';
import 'api_client.dart';
import '../models/aviso.dart';

class PublicacionesService {
  static Future<List<Aviso>> fetchPublicaciones({
    int page = 1,
    int limit = 20,
    bool? activo, // opcional: filtrar en backend si lo soporta
  }) async {
    try {
      final qp = <String, dynamic>{
        'page': page,
        'limit': limit,
        if (activo != null) 'activo': activo,
      };

      final res = await ApiClient.dio.get(
        '/publicaciones/ver-publicaciones',
        queryParameters: qp,
      );

      final body = res.data;
      final data = (body is Map) ? body['data'] : null;

      // Algunos backends devuelven data: [] o data: {rows: []}
      final List list = (data is List)
          ? data
          : (data is Map && data['rows'] is List)
              ? (data['rows'] as List)
              : const [];

      return list
          .whereType<Map>()
          .map((e) => AvisoFromApi.fromPublicacionApi(
                Map<String, dynamic>.from(e),
              ))
          .toList();
    } on DioException catch (e) {
      final msg = e.response?.data?['message'] ??
          e.response?.data?['error'] ??
          'Error al cargar publicaciones';
      throw Exception(msg);
    }
  }
}

File: src/services/publicacionesService.js
import api from '../api/axios';

const publicacionesService = {
  getAll: async (page = 1, limit = 10) => {
    const response = await api.get('/publicaciones/ver-publicaciones', {
      params: { page, limit }
    });
    return response.data;
  },

  create: async (formData) => {
    // Expects FormData object for file upload
    const response = await api.post('/publicaciones/publicaciones-crear', formData, {
      headers: {
        'Content-Type': 'multipart/form-data',
      },
    });
    return response.data;
  },

  delete: async (id) => {
    const response = await api.delete(`/publicaciones/${id}`);
    return response.data;
  },

  update: async (id, formData) => {
     const response = await api.put(`/publicaciones/${id}`, formData, {
      headers: { 'Content-Type': 'multipart/form-data' },
    });
    return response.data;
  }
};

export default publicacionesService;
